QEfficient/generation/cloud_infer.py

```diff
@@ -58,9 +58,13 @@ def __init__(
         :enable_debug_logs: bool. If True, It will enable debug logs. Default=False.
         """
         # Load QPC
-        devices = qaicrt.QIDList(device_ids)
-        self.context = qaicrt.Context(devices)
-        self.queue = qaicrt.Queue(self.context, device_ids[0])  # Async API
+        if device_ids is not None:
+            devices = qaicrt.QIDList(device_ids)
+            self.context = qaicrt.Context(devices)
+            self.queue = qaicrt.Queue(self.context, device_ids[0])
+        else:
+            self.context = qaicrt.Context()
+            self.queue = qaicrt.Queue(self.context, 0)  # Async API
```

Comment:
can you please explain, does this choose device id = 1, 2, 3, 4 when 0 is busy?

Reply:
Also, should we make `device_id` parameter optional in CLI and python APIs with this change?

Reply:
Done